modules/ggipc-client/src/client.c

```diff
@@ -272,6 +302,12 @@ GglError ggipc_call(
         }
     }
 
+    if (common_headers.message_type == EVENTSTREAM_APPLICATION_ERROR) {
+        *remote_err = GGL_ERR_FAILURE;
```

Comment:
remote_err should be set according to IPC err response

Reply:
Maybe it should be an IPC error code enum instead of GglError?

Reply:
We can always parse the error response here, log the code and message, and return the code enum value

Reply:
Another option would be to alloc the message in the passed-in allocator and return the  IpcError type